### PR TITLE
[BEAM-5845] Split Dataflow run of examples precommit to its own job

### PR DESCRIPTION
**Pull request by kennknowles**
This makes a separate Jenkins job for the most expensive pieces of the Java PreCommit uber job.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by kennknowles** [comment]
R: @swegner @alanmyrvold any pitfalls?

File: .test-infra/jenkins/job_PreCommit_Java_Examples_Dataflow.groovy [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import PrecommitJobBuilder
+
+PrecommitJobBuilder builder = new PrecommitJobBuilder(
+    scope: this,
+    nameBase: 'Java Examples Dataflow',
+    gradleTask: ':beam-runners-google-cloud-dataflow-java-examples:preCommit',
```
Comment (swegner) [inline .test-infra/jenkins/job_PreCommit_Java_Examples_Dataflow.groovy]: For other jobs we create a top-level `xyzPrecommit` task in the root build script. I don't know how the pattern was established, but it has some benefits:

1. Less logic in the Jenkins jobs, which aren't scoped to a commit and are more difficult to change. This way we can always refactor the project tree without messing with Jenkins.
1. Better discoverability, and easier to type than the super-long project name

File: examples/java/build.gradle [hunk 2]
```diff
@@ -123,7 +123,5 @@ task preCommit() {
   for (String runner : preCommitRunners) {
     dependsOn runner + "PreCommit"
   }
-  dependsOn ":beam-runners-google-cloud-dataflow-java-examples:preCommit"
-  dependsOn ":beam-runners-google-cloud-dataflow-java-examples-streaming:preCommit"
```
Comment (swegner) [inline examples/java/build.gradle]: Is the streaming pre-commit target missing from your new job?

File: .test-infra/jenkins/job_PreCommit_Java_Examples_Dataflow.groovy (hunk 1 above)
Comment (kennknowles) [reply .test-infra/jenkins/job_PreCommit_Java_Examples_Dataflow.groovy]: Ah, will do. Conversely, I think we should remove precommit/postcommit naming from everywhere else, instead giving the suites meaningful names. That way mechanism and policy are separate, and there's one place to look (the root).

**Comment by kennknowles** [comment]
Confirmed that the target works and the jobs succeeded.

**Comment by kennknowles** [comment]
run seed job

**Comment by kennknowles** [comment]
run seed job

File: .test-infra/jenkins/job_PreCommit_Java_Examples_Dataflow.groovy [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import PrecommitJobBuilder
+
+PrecommitJobBuilder builder = new PrecommitJobBuilder(
+    scope: this,
+    nameBase: 'Java Examples Dataflow',
```
Comment (alanmyrvold) [inline .test-infra/jenkins/job_PreCommit_Java_Examples_Dataflow.groovy]:
```suggestion
    nameBase: 'Java_Examples_Dataflow',
```
From https://brentlavelle.wordpress.com/2015/09/24/dont-put-spaces-in-your-jenkins-job-names/

**Comment by alanmyrvold** [comment]
Seed job maybe failing due to @swegner's https://github.com/apache/beam/pull/6805 ?

**Comment by swegner** [comment]
Sorry for the SeedJob break; I have a fix here: https://github.com/apache/beam/pull/6814

**Comment by kennknowles** [comment]
run seed job

**Comment by kennknowles** [comment]
run seed job

**Comment by kennknowles** [comment]
Looks like the seed is happy and the job exists. Thanks for the review!

**Comment by kennknowles** [comment]
Ah, I will remove the spaces. I don't believe it actually causes a problem as it happens.

**Pull request merged**